apps/spruce/src/pages/version/TestAnalysis/index.tsx

```diff
@@ -0,0 +1,235 @@
+                disabled={!hasResults}
+                id="test-failure-search-input"
+                label="Search Test Failures"
+                onChange={(value) => {
+                  setTestName(value);
+                }}
+                placeholder="Search failed tests (regex)"
+                validator={validateRegexp}
+              />
+            </div>
+            <Combobox
+              disabled={!hasResults}
+              label="Failure type"
+              multiselect
+              onChange={setSelectedTaskStatuses}
+              placeholder="Select a task status"
+              value={selectedTaskStatuses}
+            >
+              {taskStatuses.map((status) => (
+                <ComboboxOption
+                  key={status}
+                  displayName={taskStatusToCopy[status]}
+                  value={status}
+                />
+              ))}
+            </Combobox>
+            <Combobox
+              disabled={!hasResults}
+              label="Build Variant"
+              multiselect
+              onChange={setSelectedBuildVariants}
+              overflow="scroll-x"
+              placeholder="Select a build variant"
+              value={selectedBuildVariants}
+            >
+              {buildVariants.map((bv) => (
+                <ComboboxOption
+                  key={bv.buildVariant}
+                  displayName={bv.buildVariantDisplayName}
+                  value={bv.buildVariant}
+                />
+              ))}
+            </Combobox>
+          </FilterContainer>
+          {hasResults && (
+            <FilterSubheaderContainer>
+              <Body weight="medium">
+                {totalFilteredTestCount}/{totalTestCount} Failed{" "}
+                {pluralize("Test", totalFilteredTestCount)}
+              </Body>
+              <Button
+                disabled={hasFiltersApplied}
+                onClick={() => {
+                  setSelectedTaskStatuses([]);
+                  setSelectedBuildVariants([]);
+                  setTestName("");
+                }}
+                size="xsmall"
+              >
+                Clear Filters
+              </Button>
+            </FilterSubheaderContainer>
+          )}
+
+          <GroupedTestMapList groupedTestsMapEntries={groupedTestsMapEntries} />
+          {!hasMatchingResults && (
+            <BasicEmptyState
+              description="For additional analytics on tests please visit Honeycomb"
+              title={`No ${hasResults ? "Matching " : ""}Failed Tests Found`}
+            />
+          )}
+        </>
+      )}
+    </Container>
+  );
+};
+
+const Container = styled.div`
+  margin: 0 ${size.xs};
+`;
+
+const FilterSubheaderContainer = styled.div`
+  display: flex;
+  margin-bottom: ${size.m};
+  gap: ${size.xs};
+`;
+
+const FilterContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: ${size.m};
+  margin-top: ${size.xs};
+  > * {
+    width: 30%;
```

Comment:
`flex-basis` should probably be preferred to set an item's base width
```suggestion
    flex-basis: 30%;
```

Reply:
Bump

Reply:
I totally made this change https://github.com/evergreen-ci/ui/pull/407/commits/3e32d37b2240c805af970f90539c378d71ef7628#diff-4f8210ab3951ea08d400d2c6b4fcfc866f3b71b3d660a72bfe49a40085a2c311R225 but I was burned by my own revert 😞 Thanks for the follow up!